source/solvers/gls_navier_stokes.cc

```diff
@@ -818,6 +823,528 @@ GLSNavierStokesSolver<dim>::assembleGLS()
+          fe_values[velocities].get_function_gradients(
+            evaluation_point, present_velocity_gradients);
+          fe_values[velocities].get_function_laplacians(
+            evaluation_point, present_velocity_laplacians);
+
+          // Gather pressure (values, gradient)
+          fe_values[pressure].get_function_values(evaluation_point,
+                                                  present_pressure_values);
+          fe_values[pressure].get_function_gradients(
+            evaluation_point, present_pressure_gradients);
+
+          std::vector<Point<dim>> quadrature_points =
+            fe_values.get_quadrature_points();
+
+          // Calculate forcing term if there is a forcing function
+          if (l_forcing_function)
+            l_forcing_function->vector_value_list(quadrature_points, rhs_force);
+
+          // Gather the previous time steps depending on the number of stages
+          // of the time integration scheme
+          if (scheme !=
+              Parameters::SimulationControl::TimeSteppingMethod::steady)
+            fe_values[velocities].get_function_values(this->solution_m1,
+                                                      p1_velocity_values);
+
+          if (time_stepping_method_has_two_stages(scheme))
+            fe_values[velocities].get_function_values(this->solution_m2,
+                                                      p2_velocity_values);
+
+          if (time_stepping_method_has_three_stages(scheme))
+            fe_values[velocities].get_function_values(this->solution_m3,
+                                                      p3_velocity_values);
+          // Loop over the quadrature points
+          for (unsigned int q = 0; q < n_q_points; ++q)
+            {
+              // Calculation of the equivalent density at the quadrature point
+              double density_eq =
+                calculate_point_property(phase_values[q],
+                                         physical_properties.fluids[0].density,
+                                         physical_properties.fluids[1].density);
+
+              double density_eq_m1 =
+                calculate_point_property(phase_values_m1[q],
+                                         physical_properties.fluids[0].density,
+                                         physical_properties.fluids[1].density);
+
+              double dynamic_viscosity_eq = calculate_point_property(
+                phase_values[q],
+                physical_properties.fluids[0].dynamic_viscosity,
+                physical_properties.fluids[1].dynamic_viscosity);
+
+              // BB temporary
+              // Limitations for cases where air becomes liquid
+              if (density_eq < density_eq_m1)
+                density_eq_m1 = 0;
+
+              // BB temporary
+              // define epsilon below which gravity is not applied
+              double epsilon_alpha = 1e-6;
```

Comment:
Should we remove those?

Reply:
For the condition on `density_eq_m1`, right now we have `density_eq_m1 = density_eq` anyway, so it is never true. This would require further testing to consider a non-constant density as we had some issues with that, so I will remove it for now.

I think we should keep the principle of `epsilon_alpha`, so that gravity is not applied on the air part. Although, maybe we should instead consider a condition on the fluid density? If we make a simulation with two fluids that are not air, we may want to have gravity applied to both... What do you think?